### PR TITLE
fix(datepicker): datepicker fires onchange on load

### LINKED ISSUE #631: Datepicker fires onChange on render
**Describe the bug**
Datepicker fires onChange on render

**To Reproduce**
1. add Datepicker to your page
2. add console to onChange of that datepicker
3. see console on page reload/ render

**Expected behavior**
onChange should not fire on render

### PR DESCRIPTION
**Pull request by kherP**
fix #631

### CODE REVIEW

**Review by mario-subo — APPROVED** [review]
looks good to me :)

File: lib/src/Datepicker/Datepicker.tsx [hunk 1]
```diff
@@ -139,18 +142,36 @@ export const Datepicker: React.FunctionComponent<DatepickerProps> = React.forwar
...
+                const day: number = monthPicker ? 1 : customDay;
+                const month: number = customMonth;
+                const year: number = customYear;
+                const dateString: string = `${padNumber(year, true)}-${padNumber(month)}-${padNumber(day)}`;
+                const date: Date = new Date(dateString);
+                const m: number = date.getMonth() + 1;
+                if (date.getFullYear() === year && m === month && date.getDate() === day) {
+                    isDateInRange(date, min, max) ? onChange(date) : onChange(null);
+                } else {
+                    onChange(null);
```
Comment (eweseong) [inline lib/src/Datepicker/Datepicker.tsx:171]: from the code, it seems like `onChange` is still being fired on render here

File: lib/src/Datepicker/Datepicker.tsx [hunk 2]
```diff
@@ -239,6 +260,10 @@ export const Datepicker: React.FunctionComponent<DatepickerProps> = React.forwar
         const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
             const { value: changeEventValue } = e.target;
             const value: Date = new Date(changeEventValue);
+            if (min > value || max < value) {
+                onChange(null);
+                return;
+            }
             onChange(value);
```
Comment (eweseong) [inline lib/src/Datepicker/Datepicker.tsx:267]: is there a requirement to trigger `onChange` with null value? if there isn't maybe we can tap into existing helper method and trigger event only when they are valid?

```
if (isDateInRange(value, min, max)) {
  onChange(value);
}
```

File: lib/src/Datepicker/Datepicker.tsx (hunk 1 above)
Comment (kherP) [reply lib/src/Datepicker/Datepicker.tsx]: i was thinking to skip this, but added another commit for this

File: lib/src/Datepicker/Datepicker.tsx (hunk 2 above)
Comment (kherP) [reply lib/src/Datepicker/Datepicker.tsx]: updated

File: lib/src/Datepicker/Datepicker.tsx [hunk 3]
```diff
@@ -239,7 +261,11 @@ export const Datepicker: React.FunctionComponent<DatepickerProps> = React.forwar
         const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
             const { value: changeEventValue } = e.target;
             const value: Date = new Date(changeEventValue);
-            onChange(value);
+            if (isDateInRange(value, min, max)) {
+                onChange(value);
+                return;
+            }
+            onChange(null);
```
Comment (eweseong) [inline lib/src/Datepicker/Datepicker.tsx:268]: do we need to trigger change event with `null` value when the data is invalid?

**Review by eweseong — APPROVED** [review]

**Pull request merged**